src/datarobotx/idp/guard_configurations.py

```diff
@@ -0,0 +1,324 @@
+    cleaned_guard_template["name"] = guard_name
+
+    # append previous guard configurations
+    cleaned_guard_config.append(cleaned_guard_template)
+    try:
+        res = client.post(
+            "guardConfigurations/toNewCustomModelVersion/",
+            json={
+                "data": [to_api(config) for config in cleaned_guard_config],
+                "customModelId": custom_model_id,
+            },
+        )
+    except ClientError:
+        _ = get_or_create_custom_model_version_from_previous(
+            endpoint=endpoint,
+            token=token,
+            custom_model_id=custom_model_id,
+            base_environment_id=base_environment_id,
+        )
+        res = client.post(
+            "guardConfigurations/toNewCustomModelVersion/",
+            json={
+                "data": [to_api(config) for config in cleaned_guard_config],
+                "customModelId": custom_model_id,
+            },
+        )
+    return str(res.json()["customModelVersionId"])
+
+
+def get_or_create_guard_config_to_custom_model_version(  # noqa: PLR0913
+    endpoint: str,
+    token: str,
+    custom_model_id: str,
+    guard_config_template_name: str,
+    guard_config_template_settings: Dict[str, Any],
+    stages: List[Union[Literal["prompt"], Literal["response"]]],
+    intervention: Intervention,
+    name: Optional[str] = None,
+    description: Optional[str] = None,
+) -> str:
+    """Add a guard configuration to a custom model version.
+
+        If a guard configuration with the same settings already exists, it will not be created.
+
+    Parameters
+    ----------
+    endpoint : str
+        The DataRobot endpoint to connect to.
+    token : str
+        The DataRobot API token to use.
+    custom_model_id : str
+        The ID of the custom model to create the guard configuration for.
+    guard_config_template_name : str
+        The name of the guard configuration template to use.
+    guard_config_template_settings : dict
+        The settings for the guard configuration template.
+    stages : Union[List[GuardStage], List[str]]
+        The stages to apply the guard configuration to.
+    intervention : Union[Intervention, Dict[str, Any]]
+        The intervention to apply when the guard conditions are met.
+    name : Optional[str], optional
+        The name of the guard configuration, by default None
+
+    Returns
+    -------
+    str
+        The ID of the custom model version with the guard configuration.
+    """
+    return _ensure_guard_config_from_template(
+        endpoint=endpoint,
+        token=token,
+        custom_model_id=custom_model_id,
+        guard_config_template_name=guard_config_template_name,
+        guard_config_template_settings=guard_config_template_settings,
+        stages=stages,
+        intervention=intervention,
+        name=name,
+        description=description,
+        replace=False,
+    )
+
+
+def get_update_or_create_guard_config_to_custom_model_version(  # noqa: PLR0913
```

Comment:
similar to above, i found this name a bit unclear - any ideas for alternatives?
maybe:
get_update_or_create_custom_model_version_with_guard_config